From 04a7ef52fe0350156d2231d84972a51c3d1f4eab Mon Sep 17 00:00:00 2001
From: Forrest Waters <fwaters@0609.local>
Date: Wed, 31 Jul 2019 10:01:20 -0500
Subject: [PATCH] remove change that breaks older xcode versions

---
 tools/osx/xcode_locator.m | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/tools/osx/xcode_locator.m b/tools/osx/xcode_locator.m
index 1e900e4..fff0f26 100755
--- a/tools/osx/xcode_locator.m
+++ b/tools/osx/xcode_locator.m
@@ -173,14 +173,6 @@ static void AddEntryToDictionary(XcodeVersionEntry *entry,
     NSLog(@"Version strings for %@: short=%@, expanded=%@",
           url, version, expandedVersion);
 
-    NSURL *versionPlistUrl = [url URLByAppendingPathComponent:@"Contents/version.plist"];
-    NSDictionary *versionPlistContents = [[NSDictionary alloc] initWithContentsOfURL:versionPlistUrl
-                                                                               error:nil];
-    NSString *productVersion = [versionPlistContents objectForKey:@"ProductBuildVersion"];
-    if (productVersion) {
-      expandedVersion = [expandedVersion stringByAppendingFormat:@".%@", productVersion];
-    }
-
     NSURL *developerDir =
         [url URLByAppendingPathComponent:@"Contents/Developer"];
     XcodeVersionEntry *entry =
@@ -259,6 +251,12 @@ int main(int argc, const char * argv[]) {
         versionArg = @"";
       } else {
         versionArg = firstArg;
+        NSCharacterSet *versSet =
+            [NSCharacterSet characterSetWithCharactersInString:@"0123456789."];
+        if ([versionArg rangeOfCharacterFromSet:versSet.invertedSet].length
+            != 0) {
+          versionArg = nil;
+        }
       }
     }
     if (versionArg == nil) {
-- 
2.20.1 (Apple Git-117)

