diff -urN a/src/main/native/BUILD b/src/main/native/BUILD
--- a/src/main/native/BUILD	1980-01-01 00:00:00.000000000 +0300
+++ b/src/main/native/BUILD	2022-07-11 15:54:57.874189982 +0300
@@ -63,7 +63,7 @@
     ],
     includes = ["."],  # For jni headers.
     linkopts = select({
-        "//src/conditions:darwin": ["-framework CoreServices"],
+        "//src/conditions:darwin": ["-framework CoreServices -framework IOKit"],
         "//conditions:default": [],
     }),
     linkshared = 1,
diff -urN a/src/main/native/unix_jni_darwin.cc b/src/main/native/unix_jni_darwin.cc
--- a/src/main/native/unix_jni_darwin.cc	1980-01-01 00:00:00.000000000 +0300
+++ b/src/main/native/unix_jni_darwin.cc	2022-07-11 15:54:07.690646751 +0300
@@ -19,7 +19,8 @@
 #include <fcntl.h>
 #include <IOKit/IOMessage.h>
 #include <IOKit/pwr_mgt/IOPMLib.h>
-#include <os/log.h>
+// #include <os/log.h>
+#define os_log_debug(A, B) do { } while(0)
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
@@ -141,6 +142,7 @@
   return queue;
 }
 
+#if 0
 // Log used for all of our anomaly logging.
 // Logging can be traced using:
 // `log stream -level debug --predicate '(subsystem == "build.bazel")'`
@@ -161,6 +163,7 @@
   }
   return log;
 }
+#endif
 
 // The macOS implementation asserts that `msg` be a string literal (not just a
 // const char*), so we cannot use a function.
