diff -urN a/src/main/native/BUILD b/src/main/native/BUILD
--- a/src/main/native/BUILD	1980-01-01 00:00:00.000000000 +0300
+++ b/src/main/native/BUILD	2022-07-11 15:54:57.874189982 +0300
@@ -63,7 +63,7 @@
     ],
     includes = ["."],  # For jni headers.
     linkopts = select({
-        "//src/conditions:darwin": ["-framework CoreServices"],
+        "//src/conditions:darwin": ["-framework CoreServices -framework IOKit"],
         "//conditions:default": [],
     }),
     linkshared = 1,
