diff -urN a/src/main/tools/logging.h b/src/main/tools/logging.h
--- a/src/main/tools/logging.h	2022-07-11 15:56:20.909001427 +0300
+++ b/src/main/tools/logging.h	2022-07-11 15:56:44.558799549 +0300
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <time.h>
+#include <inttypes.h>
 
 // see
 // http://stackoverflow.com/questions/5641427/how-to-make-preprocessor-generate-a-string-for-line-keyword
