Index: work/src/main/tools/linux-sandbox-pid1.cc
===================================================================
--- work.orig/src/main/tools/linux-sandbox-pid1.cc
+++ work/src/main/tools/linux-sandbox-pid1.cc
@@ -40,6 +40,7 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
+#include <inttypes.h>
 
 #include <string>
 
Index: work/src/main/tools/logging.h
===================================================================
--- work.orig/src/main/tools/logging.h
+++ work/src/main/tools/logging.h
@@ -42,7 +42,7 @@
       struct timespec ts;                                           \
       clock_gettime(CLOCK_REALTIME, &ts);                           \
                                                                     \
-      fprintf(stderr, "%" PRId64 ".%09ld: %s:%d: " fmt "\n",        \
+      fprintf(stderr, "%ld.%09ld: %s:%d: " fmt "\n",        \
               ((int64_t)ts.tv_sec), ts.tv_nsec, __FILE__, __LINE__, \
               ##__VA_ARGS__);                                       \
     }                                                               \
